src/kestrel_datasource_stixbundle/interface.py

```diff
@@ -90,20 +92,24 @@ def query(uri, pattern, session_id=None, config=None):
             _logger.debug(f"requesting data from path: {data_path}")
 
             if scheme == "file":
-                bundlefile = data_path
+                rawfile = data_path
                 try:
                     with open(data_path, "r") as f:
                         bundle_in = json.load(f)
                 except Exception:
                     _clean_ingestdir_and_raise_error(ingestdir, uri)
 
             elif scheme == "http" or scheme == "https":
-                bundlefile = downloaddir / f"{data_path_stripped}.json"
-                data_uri = f"{scheme}://{data_path}"
+                data_path, dot, extension = data_path.rpartition(".")
+                data_path_stripped = "".join(filter(str.isalnum, data_path))
+                rawfile = downloaddir / f"{data_path_stripped}"
+                if extension:
+                    rawfile = rawfile.with_suffix(f".{extension}")
+                data_uri = f"{scheme}://{data_path}{dot}{extension}"
```

Comment:
Maybe a more robust way is the previous `data_uri = f"{scheme}://{data_path}"`, since the `rpartition()` may not work as expected.